src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { Person } from '../types/Person';
+import debounce from 'lodash.debounce';
+import { peopleFromServer } from '../data/people';
+
+interface AutocompleteProps {
+  onSelected: (person: Person | null) => void;
+  debounceDelay?: number;
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = React.memo(
+  ({ onSelected, debounceDelay = 1300 }) => {
+    const [query, setQuery] = useState('');
+    const [appliedQuery, setAppliedQuery] = useState('');
+    const [showList, setShowList] = useState(false);
+
+    const applyQuery = useCallback(
+      debounce(setAppliedQuery, debounceDelay),
+      [],
+    );
+
```

Comment:
Fix it 
<img width="946" alt="image" src="https://github.com/mate-academy/react_autocomplete/assets/64750958/49d864bf-cb67-4283-9e85-3dabcfcb165b">